theano/gpuarray/opt.py

```diff
@@ -215,6 +254,263 @@ def apply(self, fgraph):
+                mapping[node.outputs[0]] = node.inputs[0]
+                continue
+
+            # Move only if any of the inputs are on the GPU.
+            move_to_GPU = True
+
+            '''
+            if any([isinstance(i, GpuArrayVariable) or
+                    isinstance(i, GpuArraySharedVariable)
+                    for i in [mapping[v] for v in node.inputs] +
+                              node.outputs]):
+
+                move_to_GPU = True
+            '''
+
+            out_clients = [o.clients for o in node.outputs]
+
+            context_name = None
+            for i in [mapping[i] for i in node.inputs]:
+                if isinstance(i.type, GpuArrayType):
+                    context_name = i.type.context_name
+                    move_to_GPU = True
+                    break
+
+            new_ops = None
+            outputs = []
+            # Apply the lifter
+            for lopt in (self.local_optimizers_all +
+                         self.local_optimizers_map.get(type(node.op), []) +
+                         self.local_optimizers_map.get(node.op, [])):
+                
+                process_count[lopt] += 1
+                if move_to_GPU:
+                    node_created[lopt] += len(theano.gof.graph.ops([mapping[i] for i in node.inputs], node.outputs))
+                    t_opt = time.time()
+                    try:
+                        new_ops = lopt.transform(
+                            node.op, context_name,
+                            [mapping[i] for i in node.inputs])
+                    except TypeError:
+                        new_ops = lopt.transform(node.op, context_name, 
+                            [mapping[i] for i in node.inputs],
+                            node.outputs)
+                    finally:
+                        time_opts[lopt] += time.time() - t_opt
+                        self.new_opts.append(lopt)
+                    if new_ops:
+                        break
+            if not new_ops:
+                newnode = node.clone_with_new_inputs([mapping.get(i)
+                                                      for i in node.inputs])
+                outputs = newnode.outputs
+            elif isinstance(new_ops, (tuple, list)):
+                outputs = []
+                for o in new_ops:
+                    if o.owner and isinstance(o.owner.op, HostFromGpu):
+                        outputs.append(o.owner.inputs[0])
+                    else:
+                        outputs.append(o)
+            elif isinstance(new_ops, theano.Variable):
+                if new_ops.owner and isinstance(new_ops.owner.op, HostFromGpu):
+                    outputs = new_ops.owner.inputs
+                else:
+                    outputs = [new_ops]
+            else:
+                outputs = new_ops(*[mapping[i] for i in node.inputs],
+                                  return_list=True)
+            for new_o, old_o in zip(outputs, node.outputs):
```

Comment:
I think to compute the number of new nodes created, you need to do this before the for:
```
if new_ops:
    node_created[lopt] += len(theano.gof.graph.ops([mapping[i] for i in node.inputs], outputs))
```